File: 总题库/680.ValidPalindrome.cpp
// 贪心算法
class Solution {
public:
    bool checkPalindrome(const string &s, int low, int high) {
        for (int i = low, j = high; i <= j; i++, j--) {
            if (s[i] != s[j]) {
                return false;
            }
        }
        return true;
    }

    bool validPalindrome(string s) {
        int sLen = s.length();
        int low = 0, high = sLen - 1;
        while (low <= high) {
            if (s[low] == s[high]) {
                ++low;
                --high;
            }
            else {
                return checkPalindrome(s, low + 1, high) || checkPalindrome(s, low, high - 1);
            }
        }
        return true;
    }
};

File: 总题库/680.ValidPalindrome.py
class Solution:
    def checkPalindrome(self, s: str, low: int, high: int) -> bool:
        i = low
        j = high
        while i <= j:
            if s[i] != s[j]:
                return False
            i += 1
            j -= 1
        
        return True

    def validPalindrome(self, s: str) -> bool:
        low = 0
        high = len(s) - 1

        while low <= high:
            if s[low] == s[high]:
                low += 1
                high -= 1
            else:
                return self.checkPalindrome(s, low + 1, high) or self.checkPalindrome(s, low, high - 1)
        
        return True